src/components/icon-button/icon-button.pw.tsx

```diff
@@ -0,0 +1,232 @@
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    const iconLocator = page.getByTestId("icon").locator("..");
+    await iconLocator.focus();
+    await iconLocator.blur();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onFocus callback when a focus event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onFocus={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    const iconLocator = page.getByTestId("icon").locator("..");
+    await iconLocator.focus();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onMouseEnter callback when a mouseover event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onMouseEnter={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    const iconLocator = page.getByTestId("icon").locator("..");
+    await iconLocator.hover();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onMouseLeave callback when a mouseout event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onMouseLeave={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    const iconLocator = page.getByTestId("icon").locator("..");
+
+    await iconLocator.hover();
+    await page.mouse.move(100, 0);
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onClick callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    const iconLocator = page.getByTestId("icon").locator("..");
+    await iconLocator.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  ["Enter", "Space"].forEach((key) => {
+    test(`should call onClick callback when ${key} key is triggered`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <IconButtonComponent
+          onClick={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+      const iconLocator = page.getByTestId("icon").locator("..");
+      await iconLocator.press(key);
+      expect(callbackCount).toBe(1);
+    });
+  });
+});
+
+test.describe("check accessibility tests for IconButton component", () => {
+  test(`should pass accessibility tests for aria-label prop`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconButtonComponent aria-label={CHARACTERS.STANDARD} />);
+
+    await checkAccessibility(page);
+  });
+
+  test(`should pass accessibility tests for children prop`, async ({
```

Comment:
**suggestion** (non-blocking): i know it was written like this in the cypress test, but i'd change this test description. There is no `children` prop and in the test there's only one child. Something like `should pass accessibility tests when rendered with a child` should do